--- a/sqlite3.c
+++ b/sqlite3.c
@@ -36724,7 +36724,8 @@
   ** huge problem. Incorrectly concluding that the GNU version is available
   ** could lead to a segfault though.
   */
-#if defined(STRERROR_R_CHAR_P) || defined(__USE_GNU)
+#if (defined(STRERROR_R_CHAR_P) || defined(__USE_GNU)) \
+  && !defined(ANDROID) && !defined(__ANDROID__)
   zErr =
 # endif
   strerror_r(iErrno, aErr, sizeof(aErr)-1);
